rdf_differ/diff_getter.py

```diff
@@ -0,0 +1,242 @@
+  }
+  {
+    select ?insertionsGraph (count(*) as ?triplesInInsertionGraph)    
+    {
+      graph ?insertionsGraph {?s ?p ?o}
+    } group by ?insertionsGraph 
+  }
+}
+"""
+
+QUERY_DELETIONS_COUNT = """
+SELECT ?deletionsGraph ?triplesInDeletionGraph ?versionGraph
+WHERE {
+  graph ?versionGraph {
+    ?deletionsGraph a skos-history:SchemeDeltaDeletions .
+  }
+  {
+    select ?deletionsGraph (count(*) as ?triplesInDeletionGraph)    
+    {
+      graph ?deletionsGraph {?s ?p ?o}
+    } group by ?deletionsGraph 
+  }
+}
+"""
+
+
+class FusekiDiffGetter(AbstractDiffGetter):
+
+    def count_inserted_triples(self, dataset_name: str) -> int:
+        query_result = self.execute_query(dataset_name=dataset_name,
+                                          sparql_query=SKOS_HISTORY_PREFIXES + QUERY_INSERTIONS_COUNT)
+        return int(self._extract_insertion_count(query_result))
+
+    def count_deleted_triples(self, dataset_name: str) -> int:
+        query_result = self.execute_query(dataset_name=dataset_name,
+                                          sparql_query=SKOS_HISTORY_PREFIXES + QUERY_DELETIONS_COUNT)
+        return int(self._extract_deletion_count(query_result))
+
+    def diff_description(self, dataset_name: str) -> dict:
+        """
+            Provide a generic description
+        :param dataset_name:
+        :return:
+            * datasetURI = dataset/scheme URI,
+            * version history graph = the graph that stores the structure of the calculated diffs
+            * current version graph = the graph that corresponds to the latest version of the dataset,
+            * datasetVersions = list of loaded dataset versions as declared by the datasets themselves,
+            * versionIds = list of versionIds as provided in the configurations file
+            * versionNamedGraphs = named graphs where the versions of datasets are loaded
+        """
+        query_result = self.execute_query(dataset_name=dataset_name,
+                                          sparql_query=SKOS_HISTORY_PREFIXES + QUERY_DATASET_DESCRIPTION)
+
+        return self._extract_dataset_description(response=query_result)
+
+    def execute_query(self, dataset_name: str, sparql_query: str) -> dict:
+        endpoint = SPARQLWrapper(self.make_sparql_endpoint(dataset_name=dataset_name))
+
+        endpoint.setQuery(sparql_query)
+        endpoint.setReturnFormat(JSON)
+        return endpoint.query().convert()
+
+    def list_datasets(self) -> List[str]:
+        response = requests.get(urllib.parse.urljoin(self.triplestore_service_url, "/$/datasets"),
+                                auth=HTTPBasicAuth('admin', 'admin'))
+
+        if response.status_code != 200:
+            raise Exception(f"Fuseki server request ({response.url}) got response {response.status_code}")
+
+        return self._select_dataset_names_from_fuseki_response(response=response)
+
+    def __init__(self, triplestore_service_url: str):
```

Comment:
for readability purposes, this method should be at the beginning of the class definition